### PR TITLE
User geolocation

### PR DESCRIPTION
**Pull request by ldups**
This PR:
- uses the geolocation API to get the user's position
- shows the user position on the map
- shows the currently selected station on the map
- finds the closest station to the user if location is provided, and autofills the search by station field with that station upon initialization

Here is what the icons look like on the map:
image

I had to move some state around (up) as part of this PR. I also wanted the app to show the trains going through the closest station upon load, however to do so I had to revert the selected station to being responsive. I am not sure this makes sense from a user perspective, and I can change it back and just skip the auto-load. *Edit: removed this part*

I also added route highlighting as part of this PR. When a user selects a route, it will be highlighted on the map.

image

### CODE REVIEW

File: train-tracker/src/Search.js [hunk 1]
```diff
@@ -29,6 +28,10 @@ function Search({searchFun, routes, stations}){
         searchFun(selectedNumber, selectedRoute, selectedStation, upcoming, fromStation, toStation);
     }
 
+    useEffect(()=>{
+        searchFun(selectedNumber, selectedRoute, selectedStation, upcoming, fromStation, toStation);
+    }, [selectedStation]);
```
Comment (tjtemple) [inline train-tracker/src/Search.js:33]: are we sure we want changing station to automatically search? seems to have the side effect of having the clear button also search and is a bit inconsistent with the rest of the search only updating when button is pressed or form is submitted

**Review by tjtemple — APPROVED** [review]
Definitely a good case where refactoring for global variables would help a lot. We'll see what everyone else says about it

File: train-tracker/src/Search.js (hunk 1 above)
Comment (ldups) [reply train-tracker/src/Search.js]: Yes I removed this!

**Review by molly-pop — APPROVED** [review]
Nice work, Louise!

**Comment by molly-pop** [comment]
> Nice work, Louise!

I can def see how this would make the case for refactoring with globals if it would make this simpler. I also was looking through tabular/sidebar layouts and it could def help with that too (if we decide to do different URL endpoints for pages & not just for selected trains). I'm okay with either.

**Pull request merged**